pyomo/contrib/alternative_solutions/solnpool.py

```diff
@@ -9,119 +9,1026 @@
+        -------
+        str
+            The policy that is executed by the active pool.
+        """
+        return self.active_pool.policy
+
+    @property
+    def solutions(self):
+        """
+        Returns
+        -------
+        list
+            The solutions in the active pool.
+        """
+        return self.active_pool.solutions.values()
+
+    @property
+    def last_solution(self):
+        """
+        Returns
+        -------
+        Solution
+            The last solution added to the active pool.
+        """
+        return self.active_pool.last_solution
+
+    @property
+    def max_pool_size(self):
+        """
+        Returns
+        -------
+        int or None
+            The maximum pool size value for the active pool, or None if this parameter is not by this pool.
+        """
+        return getattr(self.active_pool, 'max_pool_size', None)
+
+    def to_dict(self):
+        """
+        Returns
+        -------
+        dict
+            A dictionary representation of the active pool.
+        """
+        return self.active_pool.to_dict()
+
+    def __iter__(self):
+        """
+        Yields
+        -------
+        Solution
+            The solutions in the active pool.
+        """
+        for soln in self.active_pool.solutions:
+            yield soln
+
+    def __len__(self):
+        """
+        Returns
+        -------
+        int
+            The number of solutions in the active pool.
+        """
+        return len(self.active_pool)
+
+    def __getitem__(self, soln_id):
+        """
+        Returns
+        -------
+        Solution
+            The specified solution in the active pool.
+        """
+        return self._pools[self._name][soln_id]
+
+    def add(self, *args, **kwargs):
+        """
+        Adds a solution to the active pool.
+
+        Returns
+        ----------
+        int
+            The index of the solution that is added.
+        """
+        return self.active_pool.add(*args, **kwargs)
+
+    #
+    # The following methods support the management of multiple
+    # pools within a PoolManager.
+    #
+
+    @property
+    def active_pool(self):
+        """
+        Gets the underlying active SolutionPool in PoolManager
+
+        Returns
+        ----------
+        SolutionPool
+            Active pool object
+
+        """
+        assert self._name in self._pools, f"Unknown pool '{self._name}'"
+        return self._pools[self._name]
+
+    def add_pool(self, *, name=None, policy="keep_best", as_solution=None, **kwds):
+        """
+        Initializes a new solution pool and adds it to this pool manager.
+
+        The method expects parameters for the constructor of the corresponding solution pool.
+        Supported pools are `keep_all`, `keep_best`, `keep_latest`, and `keep_latest_unique`.
+
+        Parameters
+        ----------
+        name : String
+            The name of the solution pool.  If name is already used then, then an error is generated.
```

Comment:
Is there a reason that the name needs to be a string?  Couldn't any hashable object be acceptable?